arangod/Cluster/DBServerAgencySync.cpp

```diff
@@ -113,9 +114,39 @@ Result DBServerAgencySync::getLocalCollections(
       }
     }
 
+    {
```

Comment:
Indeed, that was done "in a hurry". The necessary change was not as trivial as an if-statement, because then we're left with nullopt in case of replication1.
I addressed this in https://github.com/arangodb/arangodb/pull/20012/commits/b8db797a2c10918bf31e5d0a6f18aea34bb4bd52